### PR TITLE
Migrate to Spring Boot 2.5.2

### PR DESCRIPTION
**Pull request by jdubois**
Update to Spring Boot 2.5.2

### CODE REVIEW

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -67,11 +67,11 @@ inputs:
   jhipster-bom-repository:
     description: 'JHipster BOM repository'
     required: false
-    default: https://github.com/jhipster/jhipster-bom.git
+    default: https://github.com/jdubois/jhipster-bom.git
```
[the remark this answers is not in the document]
Comment (jdubois) [reply .github/actions/setup/action.yml:70]: I need this to test at the moment, but I will revert it before merging, of course

File: generators/server/index.js [hunk 2]
```diff
@@ -415,6 +416,9 @@ module.exports = class JHipsterServerGenerator extends BaseBlueprintGenerator {
             scriptsStorage.set('docker:db:up', `echo "Docker for db ${databaseType} not configured for application ${this.baseName}"`);
           }
         }
+        if (this.jhipsterConfig.searchEngine === ELASTICSEARCH) {
+          dockerAwaitScripts.push('echo "Waiting for Elasticsearch to start" && wait-on tcp:9200 && echo "Elasticsearch started"');
```
Comment (mshima) [inline generators/server/index.js:420]:
```suggestion
          dockerAwaitScripts.push('echo "Waiting for Elasticsearch to start" && wait-on "http-get://localhost:9200/_cluster/health?wait_for_status=green&timeout=60s" && echo "Elasticsearch started"');
```

File: generators/entity-server/templates/src/main/java/package/repository/search/EntitySearchRepository.java.ejs [hunk 3]
```diff
@@ -19,31 +19,40 @@
...
 import <%= packageName %>.domain.<%= persistClass %>;
-<%_ if (reactive) { _%>
-  <%_ if (!paginationNo) { _%>
+
+<%_ if (!paginationNo) { _%>
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/repository/search/EntitySearchRepository.java.ejs:28]:
```suggestion
  <%_ if (!reactive) { _%>
import java.util.List;
import java.util.stream.Collectors;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageImpl;
  <%_ } _%>
```

**Comment by jdubois** [comment]
It's all green, thank you @mshima !!!!

File: generators/server/templates/src/main/resources/config/application-dev.yml.ejs [hunk 4]
```diff
@@ -175,7 +175,7 @@ spring:
     database-platform: tech.jhipster.domain.util.FixedH2Dialect
     <%_ } _%>
   <%_ } _%>
-  <%_ if (databaseType === 'mongodb' || databaseType === 'cassandra' || searchEngine === 'elasticsearch') { _%>
+  <%_ if (databaseType === 'mongodb' || databaseType === 'cassandra' || (searchEngine === 'elasticsearch' && reactive)) { _%>
```
Comment (DanielFran) [inline generators/server/templates/src/main/resources/config/application-dev.yml.ejs:178]: With these changes, line 199 to 203 will never be filled.

Comment (DanielFran) [reply generators/server/templates/src/main/resources/config/application-dev.yml.ejs:178]:
```
    <%_ } else { _%>
   elasticsearch:
     rest:
       uris: http://localhost:9200
     <%_ } _%>
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/server/templates/src/main/resources/config/application-dev.yml.ejs (hunk 4 above)
Comment (DanielFran) [reply generators/server/templates/src/main/resources/config/application-dev.yml.ejs:178]: Mislead the condition, sorry ;)

**Review by DanielFran — APPROVED** [review]

**Comment by jdubois** [comment]
@DanielFran I think this is failing because this is linked to https://github.com/jhipster/jhipster-bom/pull/372 - those PR where OK before migrating to the "jhipster" org, so I guess we can merge them both?

**Comment by jdubois** [comment]
@DanielFran I forced the update of the new snapshot BOM on Maven Central, I hope it will help fix the build

**Comment by CGarces** [comment]
Thanks! 

Now I can test liquibase 4.4.1
``` html
        <!-- The liquibase version should match the one managed by
        https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/${spring-boot.version} -->
        <liquibase.version>4.3.5</liquibase.version>
```

**Pull request merged**